File: crates/sem-core/tests/fixtures/scope_test/swift/handlers.swift
func handleCreateDog(request: [String: Any]) -> Any {
    let name = request["name"] as! String
    return createDog(name: name)
}

func handleCreateCat(request: [String: Any]) -> Any {
    let name = request["name"] as! String
    return createCat(name: name)
}

func handleTransfer(request: [String: Any]) -> Int {
    let shelter = Shelter()
    let dog = Dog(name: request["name"] as! String)
    transferAnimal(animal: dog, shelter: shelter)
    return shelter.count()
}

func handleList(request: [String: Any]) -> [Any] {
    let animals = listAnimals()
    return animals
}

func validate(request: [String: Any]) -> Bool {
    guard let _ = request["name"] else {
        fatalError("name required")
    }
    return true
}

File: crates/sem-core/tests/fixtures/scope_test/kotlin/Handlers.kt
fun handleCreateDog(request: Map<String, Any>): Any {
    val name = request["name"] as String
    return createDog(name)
}

fun handleCreateCat(request: Map<String, Any>): Any {
    val name = request["name"] as String
    return createCat(name)
}

fun handleTransfer(request: Map<String, Any>): Int {
    val shelter = Shelter()
    val dog = Dog(request["name"] as String)
    transferAnimal(dog, shelter)
    return shelter.count()
}

fun handleList(request: Map<String, Any>): List<Any> {
    val animals = listAnimals()
    return animals
}

fun validate(request: Map<String, Any>): Boolean {
    if (request["name"] == null) {
        throw IllegalArgumentException("name required")
    }
    return true
}
